patterns/observers/Observer.kt

```diff
@@ -0,0 +1,76 @@
+
+// Interface for the subject(s) -> the one(s) who will be watched
+interface Subjects {
+  fun subscribe(observer: Observers)
+  fun unsubscribe(observer: Observers)
+  fun notifyObservers()
+  fun setState(state: String)
+  fun getState(): String
+}
+
+// Interface for the observer(s) -> the one(s) who will be watching
+interface Observers {
+  fun update(subject: Subjects)
+}
+
+class Subject : Subjects {
+  // creates a list of observers
+  private val observersList = ArrayList<Observers>()
```

Comment:
`ArrayList` or `MutableList`? `MutableList` might be more appropriate, given the generality of this example. Here's a thread that describes the difference succinctly: https://www.reddit.com/r/Kotlin/comments/kv15dx/difference_between_a_mutablelist_and_an_arraylist/